app_dart/test/request_handlers/check_for_waiting_pull_requests_test.dart

```diff
@@ -0,0 +1,402 @@
+
+- This pull request has changes requested. Please resolve those before re-applying the label.
+''',
+              'labelId': base64LabelId,
+            },
+          ),
+          MutationOptions(
+            document: removeLabelMutation,
+            variables: <String, dynamic>{
+              'id': prNoReviews.id,
+              'sBody':
+                  '''This pull request is not suitable for automatic merging in its current state.
+
+- Please get at least one approved review before re-applying this label. __Reviewers__: If you left a comment approving, please use the "approve" review action instead.
+''',
+              'labelId': base64LabelId,
+            },
+          ),
+          MutationOptions(
+            document: removeLabelMutation,
+            variables: <String, dynamic>{
+              'id': prMergeConflict.id,
+              'sBody':
+                  '''This pull request is not suitable for automatic merging in its current state.
+
+- Please resolve merge conflicts before re-applying this label.
+''',
+              'labelId': base64LabelId,
+            },
+          ),
+          MutationOptions(
+            document: removeLabelMutation,
+            variables: <String, dynamic>{
+              'id': prEverythingWrong.id,
+              'sBody':
+                  '''This pull request is not suitable for automatic merging in its current state.
+
+- Please resolve merge conflicts before re-applying this label.
+- Please get at least one approved review before re-applying this label. __Reviewers__: If you left a comment approving, please use the "approve" review action instead.
+- This pull request has changes requested. Please resolve those before re-applying the label.
+''',
+              'labelId': base64LabelId,
+            },
+          ),
+        ],
+      );
+    });
+  });
+}
+
+class FakeGraphQLClient implements GraphQLClient {
+  QueryResult Function(MutationOptions) _mutateResultForOptions;
+  QueryResult Function(QueryOptions) _queryResultForOptions;
+
+  @override
+  QueryManager queryManager;
+
+  @override
+  Cache get cache => throw UnimplementedError();
+
+  @override
+  Link get link => throw UnimplementedError();
+
+  final List<QueryOptions> queries = <QueryOptions>[];
+  final List<MutationOptions> mutations = <MutationOptions>[];
+
+  @override
+  Future<QueryResult> mutate(MutationOptions options) async {
+    mutations.add(options);
+    return _mutateResultForOptions(options);
+  }
+
+  @override
+  Future<QueryResult> query(QueryOptions options) async {
+    queries.add(options);
+    return _queryResultForOptions(options);
+  }
+
+  @override
+  Stream<FetchResult> subscribe(Operation operation) {
+    throw UnimplementedError();
+  }
+
+  @override
+  ObservableQuery watchQuery(WatchQueryOptions options) {
+    throw UnimplementedError();
+  }
+
+  void verifyQueries(List<QueryOptions> expectedQueries) {
```

Comment:
`verifyQueries` and `verifyMutations` have a lot of common code. Maybe we can create a function `_verifyOptions` as follows to remove the duplicate code:
```
void _verifyOptions(List<BaseOptions> actual, List<BaseOptions> expected) {
    expect(actual.length, expected.length);
    for (int i = 0; i < actual.length; i++) {
      expect(actual[i].toKey(), expected[i].toKey());
      expect(actual[i].fetchPolicy, expected[i].fetchPolicy);
    }
}

void verifyQueries(List<QueryOptions> expectedQueries) => _verifyOptions(queries, expectedQueries);
void verifyMutations(List<MutationOptions> expectedMutations)  => _verifyOptions(mutations, expectedMutations);
```

BTW, maybe also add a comment like: `toKey` encapsulates document, variables, ..., of the GraphQL query so we're verifying those through `toKey`.